frontend/src/container/CreateAlertV2/EvaluationSettings/__tests__/utils.test.ts

```diff
@@ -0,0 +1,354 @@
+				new Date('2024-01-16T10:30:00Z'),
+				new Date('2024-01-17T10:30:00Z'),
+			]);
+		});
+
+		it('should handle rrule with DTSTART', () => {
+			const date = dayjs('2024-01-20');
+			buildAlertScheduleFromRRule(FREQ_DAILY, date, NINE_AM_TIME);
+
+			// When date is provided, DTSTART is automatically added to the rrule string
+			expect(rrulestr).toHaveBeenCalledWith(
+				expect.stringMatching(/DTSTART:20240120T\d{6}Z/),
+			);
+		});
+
+		it('should handle rrule without DTSTART', () => {
+			// Test with no date provided - should use original rrule string
+			const result = buildAlertScheduleFromRRule(FREQ_DAILY, null, NINE_AM_TIME);
+
+			expect(rrulestr).toHaveBeenCalledWith(FREQ_DAILY);
+			expect(result).toHaveLength(3);
+		});
+
+		it('should handle escaped newlines', () => {
+			buildAlertScheduleFromRRule('FREQ=DAILY\\nINTERVAL=1', null, '10:30:00');
+
+			expect(rrulestr).toHaveBeenCalledWith('FREQ=DAILY\nINTERVAL=1');
+		});
+
+		it('should limit occurrences to maxOccurrences', () => {
+			const result = buildAlertScheduleFromRRule(FREQ_DAILY, null, '10:30:00', 2);
+
+			expect(result).toHaveLength(2);
+		});
+
+		it('should return null on error', () => {
+			(rrulestr as jest.Mock).mockImplementation(() => {
+				throw new Error('Invalid rrule');
+			});
+
+			const result = buildAlertScheduleFromRRule('INVALID', null, '10:30:00');
+			expect(result).toBeNull();
+		});
+	});
+
+	describe('buildAlertScheduleFromCustomSchedule', () => {
+		beforeEach(() => {
+			// Mock dayjs timezone methods
+			((dayjs as unknown) as { tz: jest.Mock }).tz = jest.fn(
+				(date?: string | Date) => {
+					const originalDayjs = jest.requireActual('dayjs');
+					const mockDayjs = originalDayjs(date || MOCK_DATE_STRING);
+					mockDayjs.startOf = jest.fn().mockReturnValue(mockDayjs);
+					mockDayjs.add = jest.fn().mockReturnValue(mockDayjs);
+					mockDayjs.date = jest.fn().mockReturnValue(mockDayjs);
+					mockDayjs.hour = jest.fn().mockReturnValue(mockDayjs);
+					mockDayjs.minute = jest.fn().mockReturnValue(mockDayjs);
+					mockDayjs.second = jest.fn().mockReturnValue(mockDayjs);
+					mockDayjs.daysInMonth = jest.fn().mockReturnValue(31);
+					mockDayjs.day = jest.fn().mockReturnValue(mockDayjs);
+					mockDayjs.isAfter = jest.fn().mockReturnValue(true);
+					mockDayjs.toDate = jest.fn().mockReturnValue(new Date(MOCK_DATE_STRING));
+					return mockDayjs;
+				},
+			);
+		});
+
+		it('should return null for missing required parameters', () => {
+			expect(
+				buildAlertScheduleFromCustomSchedule('', [], '10:30:00', 'UTC'),
+			).toBeNull();
+			expect(
+				buildAlertScheduleFromCustomSchedule('week', [], '10:30:00', 'UTC'),
+			).toBeNull();
+			expect(
+				buildAlertScheduleFromCustomSchedule('week', ['monday'], '', 'UTC'),
+			).toBeNull();
+			expect(
+				buildAlertScheduleFromCustomSchedule('week', ['monday'], '10:30:00', ''),
+			).toBeNull();
+		});
+
+		it('should generate monthly occurrences', () => {
```

Comment:
So I came to look at the tests to see what `generateMonthlyOccurrences` is generating. The tests here are not any help. They just assert very basic that it's not undefined and array. Please add tests to show what will be the generated occurences. The test should target 
- different timezones
- different months, and the boundary conditions
- how does the leap year and months with different number days handled.


The same goes for other possible repeay every

Question - why is there repeat every just for week and month in this change

Reply:
Was there any specific reason for only `weekly` and `monthly`? Or was it just the design didn't have other options (because they don't know use cases).

Reply:
Add in the same PR.

Reply:
Removed